### ISSUE TITLE
Close Affordance flag is ignored on Linux

### ISSUE THREAD

**Issue by kieferrm** [Microsoft/vscode#21205]
While testing #21004.

```typescript
vscode.window.showInformationMessage('hello world', 
            { modal: true }, 
            { title: 'A', isCloseAffordance: false } as vscode.MessageItem,
            { title: 'B', isCloseAffordance: false } as vscode.MessageItem,
            { title: 'C', isCloseAffordance: true } as vscode.MessageItem,
        );
```
correctly generates

<img width="292" alt="screen shot 2017-02-22 at 11 59 28 am" src="https://cloud.githubusercontent.com/assets/4674940/23229792/6b910bfc-f8f6-11e6-8334-2c42ea16acda.png">

while 
```typescript
vscode.window.showInformationMessage('hello world', 
            { modal: true }, 
            { title: 'A', isCloseAffordance: true } as vscode.MessageItem,
            { title: 'B', isCloseAffordance: false } as vscode.MessageItem,
            { title: 'C', isCloseAffordance: false } as vscode.MessageItem,
        );
```

incorrectly generates the same:
<img width="292" alt="screen shot 2017-02-22 at 11 59 28 am" src="https://cloud.githubusercontent.com/assets/4674940/23229879/c3b70714-f8f6-11e6-984e-009288b5453b.png">

When all close affordance flags are set to `false` a forth `Cancel` button is added which tells us that the close affordance flag is not ignore altogether.

**Comment by joaomoreno** [comment]
@kieferrm Can you check what happens when you press <kbd>Escape</kbd>? Do you get different results on each of those situations?

**Comment by kieferrm** [comment]
@joaomoreno No. When I run any combination of close affordance values in code like this, always `C` is logged when pressing <kbd>Escape</kbd>.
```typescript
vscode.window.showInformationMessage('hello world', 
            { modal: true }, 
            { title: 'A', isCloseAffordance: true } as vscode.MessageItem,
            { title: 'B', isCloseAffordance: false } as vscode.MessageItem,
            { title: 'C', isCloseAffordance: false } as vscode.MessageItem,
        ).then((result) => console.log(result));
```

**Comment by joaomoreno** [comment]
@kieferrm Just fixed #25226 which fixes your last comment about pressing <kbd>Escape</kbd>.

Closing this one as invalid, since Ubuntu (or GTK), in those screenshots, shows in orange the item that will be run by pressing <kbd>Enter</kbd>, not the one that will be run by pressing <kbd>Escape</kbd>. They could render it better...
